clang/test/Analysis/dtor-array.cpp

```diff
@@ -377,3 +377,26 @@ void directUnknownSymbol() {
 }
 
 }
+
+void testUnionDtor() {
```

Comment:
I think this is a wonderful test but I would not expect to find it in dtor-array as this has nothing to do with arrays. Would it be possible to move it to a new file? Also to add a same test with a union that does not declare a dtor?